src/components/AddActivityForm/AddActivityForm.js

```diff
@@ -1,22 +1,48 @@
-import {MdClose, MdOutlineAdd} from 'react-icons/md';
+import {nanoid} from 'nanoid';
+import {useState} from 'react';
+import {MdOutlineAdd} from 'react-icons/md';
 
 import {
   Form,
   Heading,
   StyledSelectDurationWrapper,
   StyledSelectTypeWrapper,
   Button,
-  CancelSpan,
   AddSpan,
-} from './ModalInputStyle';
+} from './AddActivityFormStyle';
+
+export default function AddActivityForm({onSetActivities}) {
+  const [formInput, setFormInput] = useState({id: '', name: '', location: '', duration: '', type: '', infos: ''});
+
+  //----- Data from Input -----
+  const handleChange = event => {
+    setFormInput({...formInput, [event.target.name]: event.target.value});
+  };
+  //----- Submithandler (add all input as object to state and closes modal after submit) -----
+  const handleSubmit = event => {
+    event.preventDefault();
+    const newInfos =
+      formInput.infos.indexOf('https://') !== 0
+        ? (formInput.infos = 'https://' + formInput.infos.trim().toLowerCase())
+        : formInput.infos.trim().toLowerCase();
+    onSetActivities(current => [
+      ...current,
+      {
+        id: nanoid(),
+        name: formInput.name.trim(),
+        location: formInput.location.trim().replace(/[^a-z]/gi, ' '),
+        duration: formInput.duration,
+        type: formInput.type,
+        infos: newInfos,
+        isFavorite: false,
+      },
+    ]);
+  };
 
-export default function ModalInput({onHandleChange, onHandleSubmit, onClose}) {
   return (
-    <Form aria-labelledby="addActivity" onSubmit={onHandleSubmit}>
+    <Form aria-labelledby="addActivity" onSubmit={event => handleSubmit(event)}>
```

Comment:
In this case you can also just write:
```suggestion
    <Form aria-labelledby="addActivity" onSubmit={handleSubmit}>
```
Since the `event` is the only parameter the `handleSubmit` function receives, it works like this and automatically takes the `event` as an argument.